apiserver/admin_test.go

```diff
@@ -1334,6 +1334,21 @@ func (s *macaroonLoginSuite) TestRemoteUserLoginToControllerLoginAccess(c *gc.C)
 	c.Check(result.UserInfo.Identity, gc.Equals, remoteUserTag.String())
 	c.Check(result.UserInfo.ControllerAccess, gc.Equals, "login")
 	c.Check(result.UserInfo.ModelAccess, gc.Equals, "")
+	c.Check(result.Servers, gc.DeepEquals,
+		params.FromNetworkHostsPorts(
+			parseHostPortsFromAddress(c, info.Addrs...),
+		),
+	)
+}
+
+func parseHostPortsFromAddress(c *gc.C, addresses ...string) [][]network.HostPort {
```

Comment:
Not sure there's sufficient test coverage here? Maybe it's too hard to create a k8s model and test the service address

Reply:
i was trying to add test for CAAS controller model, 
but i got some difficulty to setup a CAAS controller model because it's using `jujutesting.JujuConnSuite`..

Reply:
There's a pattern we use for that, search for 
// Set up a CAAS model to replace the IAAS one.

Reply:
I mean to setup a CAAS `controller` model which requires refactor a lot on `jujutesting.JujuConnSuite` itself.